gprofiler/merge.py

```diff
@@ -270,11 +269,12 @@ def merge_profiles(
         if process_perf is not None and perf_samples_count > 0 and ProfilingErrorStack.is_error_stack(profile.stacks):
             # runtime profiler returned an error stack; extend it with perf profiler stacks for the pid
             profile.stacks = ProfilingErrorStack.attach_error_to_stacks(process_perf.stacks, profile.stacks)
-        else:
+        elif perf_samples_count > 0:
```

Comment:
The sample counts are normalized based on system perf data here. Preserving language specific profiler samples when perf_sample_count is 0 should be done only if both perf and profilers run on the same sampling frequency. Otherwise, the data accuracy in the result may be impacted. It looks they are set to use the same sampling frequency by default (11Hz). If they can be different, we need to revisit this logic in future.